src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantCallerInternal.java

```diff
@@ -0,0 +1,338 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.util.SequenceUtil;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.vcf.VCFConstants;
+import org.apache.commons.collections4.IterableUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+import scala.Tuple3;
+import scala.Tuple4;
+import scala.Tuple5;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.BreakpointAllele.BreakpointAlleleInversion;
+import static org.broadinstitute.hellbender.tools.spark.sv.BreakpointAllele.BreakpointAlleleInversion.InversionType.INV_3_TO_5;
+import static org.broadinstitute.hellbender.tools.spark.sv.BreakpointAllele.BreakpointAlleleInversion.InversionType.INV_5_TO_3;
+
+/**
+ * Internal caller for calling structural variants.
+ */
+class SVVariantCallerInternal implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private static final Logger log = LogManager.getLogger(SVVariantCallerInternal.class);
+
+
+    /**
+     * First step in calling variants: parse all alignment records for a single locally-assembled contig and generate
+     * chimeric alignments if available.
+     *
+     * @param contigSequence                the assembled sequence
+     * @param alignmentRegionsIterable      alignment records given by aligner for the sequence
+     * @return                              the chimeric alignments of this sequence (empty if the sequence does not have any alignments)
+     */
+    static Iterable<ChimericAlignment> assembleBreakpointsFromAlignmentRegions(final byte[] contigSequence, final Iterable<AlignmentRegion> alignmentRegionsIterable) {
+        final List<AlignmentRegion> alignmentRegions = IterableUtils.toList(alignmentRegionsIterable);
+        if (alignmentRegions.size() > 1) { // todo: should remove this "if checking" and always sort when the input is guaranteed to be of size > 1
```

Comment:
Yep. So just to be sure, currently we don't want to make use of locally assembled contigs that have a simple nice unique alignment for calling. Correct?